src/general/application/content-types/content-types-repository.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+// phpcs:disable Yoast.NamingConventions.NamespaceName.TooLong
+namespace Yoast\WP\SEO\General\Application\Content_Types;
+
+use Yoast\WP\SEO\General\Application\Taxonomies\Taxonomies_Repository;
+use Yoast\WP\SEO\General\Domain\Content_Types\Content_Type;
+use Yoast\WP\SEO\Helpers\Post_Type_Helper;
+
+/**
+ * The repository to get content types.
+ */
+class Content_Types_Repository {
+
+	/**
+	 * The post type helper.
+	 *
+	 * @var Post_Type_Helper
+	 */
+	protected $post_type_helper;
+
+	/**
+	 * The taxonomies repository.
+	 *
+	 * @var Taxonomies_Repository
+	 */
+	private $taxonomies_repository;
+
+	/**
+	 * The constructor.
+	 *
+	 * @param Post_Type_Helper      $post_type_helper      The post type helper.
+	 * @param Taxonomies_Repository $taxonomies_repository The taxonomies repository.
+	 */
+	public function __construct(
+		Post_Type_Helper $post_type_helper,
+		Taxonomies_Repository $taxonomies_repository
+	) {
+		$this->post_type_helper      = $post_type_helper;
+		$this->taxonomies_repository = $taxonomies_repository;
+	}
+
+	/**
+	 * Returns the content types object.
+	 *
+	 * @return array<string, string|array<string,string|array<string, string>>|null> The content types object.
+	 */
+	public function get_content_types(): array {
+		$content_types = [];
+		$post_types    = $this->post_type_helper->get_indexable_post_types();
+
+		foreach ( $post_types as $post_type ) {
```

Comment:
I think you might want to introduce another domain object here that replaces that `$content_types` array with an Content_Types_List object that holds all these content types. This way you ensure only valid Content_Types can be added there (by using typing in an `add` method) And that List object can have a ->to_json or ->to_array esq method if needed for the script data.

Reply:
I'm interested to hear what's the purpose of that change, what are the advantages? Because I'm not seeing it 👀

Reply:
A great follow-up call made this clearer to me!